torchtext/datasets/iwslt2016.py

```diff
@@ -0,0 +1,331 @@
+import os
+import io
+import codecs
+import xml.etree.ElementTree as ET
+from torchtext.utils import (download_from_url, extract_archive)
+from torchtext.data.datasets_utils import RawTextIterableDataset
+from torchtext.data.datasets_utils import wrap_split_argument
+
+
+SUPPORTED_DATASETS = {
+    'URL': 'https://drive.google.com/uc?id=1l5y6Giag9aRPwGtuZHswh3w5v3qEz8D8',
+    '_PATH': '2016-01.tgz',
+    'MD5': 'c393ed3fc2a1b0f004b3331043f615ae',
+    'valid_test': ['dev2010', 'tst2010', 'tst2011', 'tst2012', 'tst2013', 'tst2014'],
+    'language_pair': {
+        'en': ['ar', 'de', 'fr', 'cs'],
+        'ar': ['en'],
+        'fr': ['en'],
+        'de': ['en'],
+        'cs': ['en'],
+    },
+    'year': 16,
+
+}
+
+URL = SUPPORTED_DATASETS['URL']
+MD5 = SUPPORTED_DATASETS['MD5']
+
+NUM_LINES = {
+    'train': {
+        'train': {
+            ('en', 'ar'): 224126,
```

Comment:
If you use sets here as a key you don't need to store both orders of language pairs